models/contacts.js

```diff
@@ -1,14 +1,101 @@
-// const fs = require('fs/promises')
+const fs = require('fs/promises');
+const path = require('path');
+const color = require('colors');
+const { nanoid } = require('nanoid');
 
-const listContacts = async () => {}
+const pathContacts = path.join(__dirname, 'contacts.json');
+console.log(pathContacts);
 
-const getContactById = async (contactId) => {}
+const readContactsFile = async()=> {
+  try{
+    const data = await fs.readFile(pathContacts, 'utf-8');
+    console.log('Contacts successfully read!');
+    return JSON.parse(data);
+  }catch(err){
+    console.error('Error reading contacts:', err.message);
+    return [];
+  }
+}
+// readContactsFile()
+// .then(
+//   contacts => {console.log(contacts)}
+// )
+
+const writeContactsFile = async (contacts) => {
+  try {
+    await fs.writeFile(pathContacts, JSON.stringify(contacts, null, 2));
+    console.log('Contacts successfully written to file!'.green);
+  } catch (err) {
+    console.error('Error writing contacts to file!', err.red);
+  }
+};
+
+
+const listContacts = async () => {
+  const contacts = await readContactsFile();
+  console.log('Contacts successfully returned!'.blue);
+  console.table(contacts);
+  return contacts;
+}
+const getContactById = async (contactId) => {
+  const contacts = await readContactsFile();
+  const contactById = contacts.find(contact => contact.id === contactId);
+  console.log(`Contact with ID ${contactId} is displayed!`.blue);
+  return contactById;
+}
+
+const removeContact = async (contactId) => {
+  const contacts = await readContactsFile();
```

Comment:
you can wrap this in a try catch, the delete operation can fail at DB level so it would be nice to have some error handling.